pkg/controllers/deprovisioning/suite_test.go

```diff
@@ -1006,10 +1035,14 @@ var _ = Describe("Delete Node", func() {
 		ExpectReconcileSucceeded(ctx, deprovisioningController, client.ObjectKey{})
 		wg.Wait()
 
+		// Cascade any deletion of the machine to the node
+		ExpectMachinesCascadeDeletion(ctx, env.Client, machine1)
+
 		// we don't need a new node
+		Expect(ExpectMachines(ctx, env.Client)).To(HaveLen(1))
```

Comment:
I worry a little bit about how every test now has to contend with the complexity of both machines and nodes.

Reply:
Agreed. Good callouts, will handle in a follow-up PR